stan/math/prim/fun/value_of.hpp

```diff
@@ -24,7 +24,7 @@ namespace math {
  * @param x scalar to convert to double
  * @return value of scalar cast to double
  */
-template <typename T>
+template <typename T, require_arithmetic_t<T>* = nullptr>
```

Comment:
Doesn't the current specification return ```double``` and ```int``` unchanged, and now you're casting both to ```double```?

Reply:
It does and I am not. I leaft in specializations for int and double, which return arguments unchanged.